timeseries/src/autogluon/timeseries/models/chronos/model.py

```diff
@@ -0,0 +1,366 @@
+        hyperparameters = hyperparameters if hyperparameters is not None else {}
+        self.batch_size = hyperparameters.get("batch_size", self.default_batch_size)
+        self.num_samples = hyperparameters.get("num_samples", self.default_num_samples)
+        self.model_path = hyperparameters.get("model_path", self.default_model_path)
+        self.skip_validation = hyperparameters.get("skip_validation", False)
+        self.device = hyperparameters.get("device", None)
+        self.optimization_strategy: Optional[Literal["onnx", "ovm"]] = hyperparameters.get(
+            "optimization_strategy", None
+        )
+        self.context_length = hyperparameters.get("context_length", self.default_context_length)
+
+        model_path_safe = str.replace(self.model_path, "/", "__")
+        name = (name if name is not None else "Chronos") + f"[{model_path_safe}]"
+
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+
+        self.model_pipeline: Optional[OptimizedChronosPipeline] = None
+
+    def save(self, path: str = None, verbose: bool = True) -> str:
+        pipeline = self.model_pipeline
+        self.model_pipeline = None
+        path = Path(super().save(path=path, verbose=verbose))
+        self.model_pipeline = pipeline
+
+        return str(path)
+
+    @classmethod
+    def load(cls, path: str, reset_paths: bool = True, verbose: bool = True) -> "ChronosModel":
+        model = load_pkl.load(path=os.path.join(path, cls.model_file_name), verbose=verbose)
+        model.model_pipeline = model._get_model_pipeline()
+        if reset_paths:
+            model.set_contexts(path)
+        return model
+
+    def score_and_cache_oof(
+        self,
+        val_data: TimeSeriesDataFrame,
+        store_val_score: bool = False,
+        store_predict_time: bool = False,
+    ) -> None:
+        if self.skip_validation:
+            logger.info("\tSkipping validation for this model...")
+            self._oof_predictions = [None]
+            self.val_score = None
+            self.predict_time = None
+        else:
+            super().score_and_cache_oof(val_data, store_val_score, store_predict_time)
+
+    def _is_gpu_available(self) -> bool:
+        import torch.cuda
+
+        return torch.cuda.is_available()
+
+    def get_minimum_resources(self, is_gpu_available: bool = False) -> Dict[str, Union[int, float]]:
+        minimum_resources = {"num_cpus": 1}
+        # if GPU is available, we train with 1 GPU per trial
+        if is_gpu_available:
+            minimum_resources["num_gpus"] = MODEL_CONFIGS[self.model_path].get("num_gpus", 0)
+        return minimum_resources
+
+    def _get_model_pipeline(self):
+        gpu_available = self._is_gpu_available()
+
+        assert self.model_path in MODEL_CONFIGS or Path.is_dir(
```

Comment:
Do I understand correctly that we only support loading weights from 1/ official Amazon HF repos or 2/ local directory?

What if we allow arbitrary HF paths using logic similar to https://github.com/allenai/cached_path?tab=readme-ov-file#usage? 
- `hf://model/path` - HF repo
- `file://model/path` or `model/path` - local path
- `small` / `tiny` / `mini` / `base` / `large` - automatically resolves to the official Amazon weights on HF?

Reply:
We can definitely implement the shorthands. I'm not sure about the value the other formats bring. Are you also suggesting that we use the `cached_path` library? (i.e., HF already provides filesystem caching out of the box)

Reply:
I didn't mean using the `cached_path` library, rather using the `hf://` and `file://` prefixes to differentiate between huggingface & local paths

Reply:
Right now HuggingFace logic seems to be to look for a local path first. If there's a clash between hf_hub and local path, then local path is preferred. I think this is highly unlikely, and would like to keep the current implementation.